engine/apps/grafana_plugin/views/sync_v2.py

```diff
@@ -25,7 +28,14 @@ class SyncV2View(APIView):
     authentication_classes = (BasePluginAuthentication,)
 
     def do_sync(self, request: Request) -> Organization:
-        serializer = SyncDataSerializer(data=request.data)
+        if request.headers.get("Content-Encoding") == "gzip":
+            gzip_data = gzip.GzipFile(fileobj=io.BytesIO(request.body)).read()
```

Comment:
Curious, can't you pass the `request` directly as the fileobj? (I think the `request` object implements file-like interface?)